repositories_test.go

```diff
@@ -0,0 +1,253 @@
+// +build !integration
+
+package function_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	fn "knative.dev/kn-plugin-func"
+)
+
+const RepositoriesTestRepo = "repository-a"
+
+// TestRepositoriesList ensures the base case of an empty list
+// when no repositories are installed.
+func TestRepositoriesList(t *testing.T) {
+	root, rm := mktemp(t)
```

Comment:
I know it's a bit pedantic, but `mktemp()` does not return a URL. It just returns a path string (which is why, I assume, you have `ss := strings.Split(uri, string(os.PathSeparator))` in `repositories.go`). I wonder if the interface should actually use a `URL` type from `net/url` to ensure proper input. In that case, this might be:

```suggestion
	root, rm := "file://" + filepath.ToSlash(mktemp(t))
```

Reply:
Not pedantic at all!  I am glad you brought it up, because this is probably more correct:

The tests which generate a local `file://` path for addition now do so using `filepath.ToSlash`.
The string passed in to `.Add` is now validated as a URL using `net/url.Parse`.

Thanks for the suggestion!  It's indeed better like this.

Regarding using URL as the input type:  I stopped short of _requiring_ input of type `net.URL` because, from what I have seen, the idiom has been to use strings in most all cases due to the notoriously loose definition of what constitutes a URL; it's requirement merely shifting the expectation of type shifting and validation to the library's user.   Instead sticking with strings for the API and doing the work ourselves of validating that no explicitly _invalid_ urls are passed, returning a standard error on invalid strings.  Thus the _user_ of the library gets the benefit of simple string acceptance and mutation without needing to import the net package, nor parse and error handle themselves.  I hope this is an acceptable compromise.

Reply:
Before I commented on this yesterday I did a bit of reading about URL vs URI and I agree, the whole thing is a mess, specified in a couple of archaic RFCs. Your approach here makes perfect sense in this context!